bundles/org.openhab.binding.bluetooth.dbusbluez/src/main/java/org/openhab/binding/bluetooth/dbusbluez/handler/DBusBlueZBridgeHandler.java

```diff
@@ -0,0 +1,343 @@
+
+        if (adapters == null || adapters.isEmpty()) {
+            return false;
+        }
+
+        for (com.github.hypfvieh.bluetooth.wrapper.BluetoothAdapter btAdapter : adapters) {
+            if (btAdapter.getAddress() != null
+                    && btAdapter.getAddress().equalsIgnoreCase(this.adapterAddress.toString())) {
+                // Found the good adapter
+                this.adapter = btAdapter;
+
+                logger.debug("Turning off adapter...");
+                // Power cycle OFF / ON for a clean start
+                this.adapter.setPowered(false);
+                logger.debug("Adapter state: {}", this.adapter.isPowered());
+
+                DBusBlueZUtils.sleep(1000);
+                logger.debug("Turning on adapter...");
+
+                this.adapter.setPowered(true);
+
+                logger.debug("Adapter DBUS path: {}", this.adapter.getDbusPath());
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    private void initializeInternal() {
+
+        // Load configuration
+        final DBusBlueZAdapterConfiguration configuration = getConfigAs(DBusBlueZAdapterConfiguration.class);
+        if (configuration.address != null) {
+            this.adapterAddress = new BluetoothAddress(configuration.address.toUpperCase());
+        } else {
+            // If configuration does not contain adapter address to use, exit with error.
+            logger.info("Adapter MAC address not provided");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "address not set");
+            return;
+        }
+
+        logger.debug("Creating BlueZ adapter with address '{}'", adapterAddress);
+
+        if (!initializeDeviceManager()) {
+            // Device manager not initialized so exiting.
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, "Library error.");
+            return;
+        }
+
+        if (!initializeAdapter()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "Bluetooth adapter could not be found.");
+            return;
+        }
+
+        // Beyond this point, lib is initialized and adapter was found.
+        updateStatus(ThingStatus.ONLINE);
+
+        this.discoveryJob = scheduler.scheduleAtFixedRate(this::refreshDevices, 10, 10, TimeUnit.SECONDS);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * This method is essential as most cheap USB chips will not supporte scanning
+     * AND connecting simultaneously. The scanning state can also be changed by
+     * the operating system or bluetoothctl command, so it is mandatory to check
+     * state before trying to connect.
```

Comment:
Isn't this something that is supposed to be handled by bluez?

Reply:
It seems it works fine now yes. So removing commend.